src/mame/hp/hp98x6_upi.h

```diff
@@ -24,6 +24,8 @@ class hp98x6_upi_device : public device_t
 	// construction/destruction
 	hp98x6_upi_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
 
+	void set_id_prom(const uint8_t *ptr) { m_idprom = ptr; }
+
```

Comment:
Rather than doing this, it’s better to have an `optional_region_ptr` to find the IDPROM.  You can either have it start with the guaranteed not to be found dummy tag and let the system instantiating it configure the tag of the ROM region, or use a tag relative to the UPI device (e.g. initialise with `m_idprom(*this, "idprom")` and use the region tag `"upi:idprom"` in the ROM definitions for the parent system).

Reply:
Ok, I think I addressed the issue you saw. Keep in mind that access to IDPROM through the UPI only happened in 9816, other systems had the PROM directly mapped on 68k bus.

Reply:
Also, to reply to @MooglyGuy, IDPROM region was marked as optional because I initially thought to replicate in emulation the behavior of real systems that could also run without ID PROMs. But in the end, ID PROM image is always present so why bother.